src/crypto-bank.WebAPI/Program.cs

```diff
@@ -1,47 +1,78 @@
 
-builder.RegisterDependencies();
+builder.Services.AddLogging(loggingBuilder => loggingBuilder.AddConsole());
 
-var app = builder.Build();
+builder.Services.AddMediatR(cfg => cfg
+    .RegisterServicesFromAssembly(Assembly.GetExecutingAssembly())
+    .AddOpenBehavior(typeof(ValidationBehavior<,>)));
 
-var logger = app.Services.GetRequiredService<ILogger<Program>>();
+builder.Services.AddSingleton<Dispatcher>();
 
-app.MapPost("/user/register",
-    async (
-        UserRegistrationRequest request,
-        UserService userService,
-        IValidator<UserRegistrationRequest> requestValidator) =>
-    {
-        logger.LogInformation($"Registering user with email [{request.Email}]");
+builder.Services
+    .AddControllers(options => options.ModelValidatorProviders.Clear())
+    .AddJsonOptions(options => options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
 
-        await requestValidator.ValidateAndThrowAsync(request);
+builder.Services.AddHttpContextAccessor();
 
-        var registeredUser = await userService.Register(request.Email, request.Password);
+builder.Services.AddScoped<ExceptionHandlerMiddleware>();
 
-        logger.LogInformation($"User was registered with id [{registeredUser.Id}]");
+builder.Services
+    .AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+    .AddJwtBearer(options =>
+    {
+        var jwtOptions = builder.Configuration.GetSection("Features:Auth").Get<AuthOptions>()!.Jwt;
 
-        return Results.Created($"/user/{registeredUser.Id}", new UserRegistrationResponse { Id = registeredUser.Id });
+        options.TokenValidationParameters = new TokenValidationParameters
+        {
+            ValidateIssuer = true,
+            ValidateAudience = true,
+            ValidateLifetime = true,
+            ValidateIssuerSigningKey = true,
+            ValidIssuer = jwtOptions.Issuer,
+            ValidAudience = jwtOptions.Audience,
+            IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(jwtOptions.SigningKey)),
+            //todo I have two places for configuring how to work with tokens. I need to refactor it
+        };
     });
 
-app.MapPost("/user/login",
-    async (UserLoginRequest request, TokenService tokenService) =>
-    {
-        logger.LogInformation($"Logging in user with email [{request.Email}]");
+builder.Services.AddAuthorization(options =>
+{
+    options.AddPolicy(PolicyName.AdministratorRole, policy => policy.RequireRole(Role.Administrator.ToString()));
+    options.AddPolicy(PolicyName.AnalystRole, policy => policy.RequireRole(Role.Analyst.ToString()));
+});
 
-        var token = await tokenService.Create(request.Email, request.Password);
+builder.Services.AddCommonProject();
+builder.Services.AddDatabaseProject();
 
-        return Results.Ok(new UserLoginResponse(token.AccessToken, token.RefreshToken));
-    });
+builder.Services.AddCommon();
+builder.Services.AddAccounts(builder.Configuration);
```

Comment:
I agree with WebAPI project, but I did it on purpose, because `WebApplicationBuilder` in `AspNetCore` namespace and I didn't want to create dependency from projects `Common` and `Database` on `Microsoft.AspNetCore`. And I wanted to make consistent all dependency extension classes.